### PR TITLE
fix(autoware_elevation_map_loader): add empty point cloud guards

### PR DESCRIPTION
**Pull request by youtalk**
## Description

Add validation to check for empty point clouds before processing to prevent undefined behavior in PCL functions and potential crashes.

- Add guard in `onPointcloudMap` callback
- Add guard in `receiveMap` after concatenation

## Related links

**Parent Issue:**

- #11737

## How was this PR tested?

- Built successfully in Docker environment with `ghcr.io/autowarefoundation/autoware:universe-devel`
- All 11 tests passed (0 errors, 0 failures)

## Notes for reviewers

The fix follows the recommended pattern from issue #11737:
```cpp
if (pointcloud_map->data.empty() || pointcloud_map->width == 0 || pointcloud_map->height == 0) {
  RCLCPP_WARN(this->get_logger(), "Empty pointcloud_map received, skipping processing");
  return;
}
```

Note: `RCLCPP_WARN` is used here instead of `RCLCPP_DEBUG` because receiving an empty map is likely an unexpected condition that should be visible to operators.

## Interface changes

None.

## Effects on system behavior

When empty point cloud maps are received, the node will now skip processing and return early instead of potentially causing undefined behavior in PCL functions.

### CODE REVIEW

File: perception/autoware_elevation_map_loader/src/elevation_map_loader_node.cpp [hunk 1]
```diff
@@ -294,6 +301,13 @@ void ElevationMapLoaderNode::receiveMap()
     concatenatePointCloudMaps(pointcloud_map, result.get()->new_pointcloud_with_ids);
   }
   RCLCPP_DEBUG(this->get_logger(), "Pointcloud map receiving process has been finished");
+
+  // check for empty point cloud
+  if (pointcloud_map.data.empty() || pointcloud_map.width == 0 || pointcloud_map.height == 0) {
+    RCLCPP_WARN(this->get_logger(), "Empty pointcloud_map received after concatenation");
+    return;
+  }
```
[the remark this answers is not in the document]
Comment (youtalk) [reply perception/autoware_elevation_map_loader/src/elevation_map_loader_node.cpp:314]: https://github.com/autowarefoundation/autoware_universe/pull/11747/commits/cf5fe19b2430cf137f7e76fd64178a9d13c0b1db

**Comment by youtalk** [comment]
@asa-naki @kosuke55 @Shin-kyoto Please review this.

File: perception/autoware_elevation_map_loader/src/elevation_map_loader_node.cpp [hunk 2]
```diff
@@ -294,9 +302,18 @@ void ElevationMapLoaderNode::receiveMap()
     concatenatePointCloudMaps(pointcloud_map, result.get()->new_pointcloud_with_ids);
   }
   RCLCPP_DEBUG(this->get_logger(), "Pointcloud map receiving process has been finished");
+
+  // check for empty point cloud
+  // TODO(youtalk): add unit test for empty point cloud handling
+  if (pointcloud_map.data.empty() || pointcloud_map.width == 0 || pointcloud_map.height == 0) {
+    RCLCPP_WARN(this->get_logger(), "Empty pointcloud_map received after concatenation");
```
Comment (Shin-kyoto) [inline perception/autoware_elevation_map_loader/src/elevation_map_loader_node.cpp]: @youtalk 
Thank you for your PR!

[IMO]
Consider using `RCLCPP_WARN_THROTTLE` to avoid cluttering the logs.

**Review by asa-naki — APPROVED** [review]
LGTM

**Review by Shin-kyoto — CHANGES_REQUESTED** [review]

File: perception/autoware_elevation_map_loader/src/elevation_map_loader_node.cpp (hunk 2 above)
Comment (youtalk) [reply perception/autoware_elevation_map_loader/src/elevation_map_loader_node.cpp]: Fixed. https://github.com/autowarefoundation/autoware_universe/pull/11747/commits/83af705c07f5af2a96584bbd0cb24964a2deed7d

**Comment by youtalk** [comment]
@Shin-kyoto Please review again.

**Review by Shin-kyoto — APPROVED** [review]
Thank you so much for your PR! LGTM

- I confirmed that this PR's quality is enough to merge.
- I did **NOT** check this PR by running autoware, just checking the code quality.
